src/Signature/SignatureV4.php

```diff
@@ -43,9 +50,13 @@ public function signRequest(
         if ($token = $credentials->getSecurityToken()) {
             $parsed['headers']['X-Amz-Security-Token'] = [$token];
         }
-
         $cs = $this->createScope($sdt, $this->region, $this->service);
         $payload = $this->getPayload($request);
+
+        if ($this->unsigned) {
```

Comment:
This was added so that we only add the header `X-Amz-Content-Sha256` to v4-unsigned requests.
Checking if the payload is `UNSIGNED_PAYLOAD` is same as checking for the boolean flag. 

Can you please help me in understanding why would it be better to check for `self::UNSIGNED_PAYLOAD` and not the simple boolean flag.